### PR TITLE
External profiler marking script

### PR DESCRIPTION
**Pull request by FlorianDeconinck**
## Purpose

Add marking for external profiler, targeting nvidia `nsys`

## Code changes:

- `profiler/external_profiler.py` as a script to call before the python run to generate the marking during runtime using the `sys.setprofile` hook of Python
- Basic coverage for fv3core marking

## Requirements changes:

N/A

## Infrastructure changes:

N/A

## Checklist
Before submitting this PR, please make sure:

- [ ] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: profiler/external_profiler.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    if (
+        frame.f_code.co_name == "__call__"
+        and frame.f_code.co_filename == "fv3core/stencils/dyn_core.py"
+    ):
+        return "Acoustic timestep"
+    elif (
+        frame.f_code.co_name == "compute"
+        and frame.f_code.co_filename == "fv3core/stencils/remapping.py"
+    ):
+        return "Remapping"
```
Comment (rheacangeo) [inline profiler/external_profiler.py:38]: Might be cool to add TracerAdvections from fv3core/stencils/tracer_2d_1l.py to match the typical plots we show, though not crucial

**Review by rheacangeo — APPROVED** [review]
This is really awesome and clean and spiffy, nice work!

**Comment by rheacangeo** [comment]
We should add your name to CONTRIBUTORS.md in one of these great PRs you are doing!

File: profiler/external_profiler.py (hunk 1 above)
Comment (FlorianDeconinck) [reply profiler/external_profiler.py:38]: Yep, just had a chat with Tobias I'll add the `TracerAdvection.__call__` and also what I can to mark halo updates

**Pull request merged**